query_test.go

```diff
@@ -49,19 +49,63 @@ func TestClient_Search(t *testing.T) {
 
 	text := randomString(10)
 
-	_, err := ch.SendMessage(&Message{Text: text + " " + randomString(25)}, user1.ID)
+	_, err := ch.SendMessage(&Message{
+		Text:      text + " 1",
+		ExtraData: map[string]interface{}{"color": "green"},
+	}, user1.ID)
 	mustNoError(t, err)
 
-	_, err = ch.SendMessage(&Message{Text: text + " " + randomString(25)}, user2.ID)
+	_, err = ch.SendMessage(&Message{
+		Text:      text + " 2",
+		ExtraData: map[string]interface{}{"color": "red"},
+	}, user2.ID)
 	mustNoError(t, err)
 
-	got, err := c.Search(SearchRequest{Query: text, Filters: map[string]interface{}{
-		"members": map[string][]string{
-			"$in": {user1.ID, user2.ID},
-		},
-	}})
+	t.Run("search message text", func(t *testing.T) {
+		got, err := c.Search(SearchRequest{Query: text, Filters: map[string]interface{}{
+			"members": map[string][]string{
+				"$in": {user1.ID, user2.ID},
+			},
+		}})
 
-	mustNoError(t, err)
+		mustNoError(t, err)
+
+		assert.Len(t, got, 2)
+		mustNoError(t, err)
+	})
+
+	t.Run("search text and custom props via message filters", func(t *testing.T) {
+		got, err := c.Search(SearchRequest{
+			Filters: map[string]interface{}{
+				"members": map[string][]string{
+					"$in": {user1.ID, user2.ID},
+				}},
+			MessageFilters: map[string]interface{}{
+				"text":  map[string]interface{}{"$q": text},
+				"color": "green",
+			},
+		})
+
+		mustNoError(t, err)
+		assert.Len(t, got, 1)
+	})
 
-	assert.Len(t, got, 2)
+	t.Run("sorting test", func(t *testing.T) {
+		got, err := c.Search(SearchRequest{
+			Filters: map[string]interface{}{
+				"members": map[string][]string{
+					"$in": {user1.ID, user2.ID},
+				}},
+			MessageFilters: map[string]interface{}{
+				"text": map[string]interface{}{"$q": text},
+			},
+			Sort: []*SortOption{{Field: "created_at", Direction: -1}},
+		})
+
+		mustNoError(t, err)
+		assert.Len(t, got, 2)
+
+		assert.Equal(t, got[0].Text, text+" 2")
```

Comment:
The order of the parameters is wrong. First expected then got.

It mightn't be seen on success but on fail, it will generate a wrong failure message.